pkg/yang/types_builtin.go

```diff
@@ -588,17 +701,30 @@ func (n Number) Less(m Number) bool {
 		return true
 	case n.Kind != Negative && m.Kind == Negative:
 		return false
-	case n.Kind == Negative:
+	}
```

Comment:
I think that check was there already. I just removed the unnecessary case which is covered by the preceding cases.